{
    "GSD": {
        "alias": "CVE-2020-15223",
        "description": "In ORY Fosite (the security first OAuth2 & OpenID Connect framework for Go) before version 0.34.0, the `TokenRevocationHandler` ignores errors coming from the storage. This can lead to unexpected 200 status codes indicating successful revocation while the token is still valid. Whether an attacker can use this for her advantage depends on the ability to trigger errors in the store. This is fixed in version 0.34.0",
        "id": "GSD-2020-15223"
    },
    "namespaces": {
        "cve.org": {
            "CVE_data_meta": {
                "ASSIGNER": "security-advisories@github.com",
                "ID": "CVE-2020-15223",
                "STATE": "PUBLIC",
                "TITLE": "Ignored storage errors on token revokation in ORY Fosite"
            },
            "affects": {
                "vendor": {
                    "vendor_data": [
                        {
                            "product": {
                                "product_data": [
                                    {
                                        "product_name": "fosite",
                                        "version": {
                                            "version_data": [
                                                {
                                                    "version_value": "< 0.34.0"
                                                }
                                            ]
                                        }
                                    }
                                ]
                            },
                            "vendor_name": "ory"
                        }
                    ]
                }
            },
            "data_format": "MITRE",
            "data_type": "CVE",
            "data_version": "4.0",
            "description": {
                "description_data": [
                    {
                        "lang": "eng",
                        "value": "In ORY Fosite (the security first OAuth2 & OpenID Connect framework for Go) before version 0.34.0, the `TokenRevocationHandler` ignores errors coming from the storage. This can lead to unexpected 200 status codes indicating successful revocation while the token is still valid. Whether an attacker can use this for her advantage depends on the ability to trigger errors in the store. This is fixed in version 0.34.0"
                    }
                ]
            },
            "impact": {
                "cvss": {
                    "attackComplexity": "HIGH",
                    "attackVector": "NETWORK",
                    "availabilityImpact": "NONE",
                    "baseScore": 8,
                    "baseSeverity": "HIGH",
                    "confidentialityImpact": "HIGH",
                    "integrityImpact": "HIGH",
                    "privilegesRequired": "NONE",
                    "scope": "CHANGED",
                    "userInteraction": "REQUIRED",
                    "vectorString": "CVSS:3.1/AV:N/AC:H/PR:N/UI:R/S:C/C:H/I:H/A:N",
                    "version": "3.1"
                }
            },
            "problemtype": {
                "problemtype_data": [
                    {
                        "description": [
                            {
                                "lang": "eng",
                                "value": "CWE-755: Improper Handling of Exceptional Conditions"
                            }
                        ]
                    }
                ]
            },
            "references": {
                "reference_data": [
                    {
                        "name": "https://github.com/ory/fosite/security/advisories/GHSA-7mqr-2v3q-v2wm",
                        "refsource": "CONFIRM",
                        "url": "https://github.com/ory/fosite/security/advisories/GHSA-7mqr-2v3q-v2wm"
                    },
                    {
                        "name": "https://github.com/ory/fosite/commit/03dd55813f5521985f7dd64277b7ba0cf1441319",
                        "refsource": "MISC",
                        "url": "https://github.com/ory/fosite/commit/03dd55813f5521985f7dd64277b7ba0cf1441319"
                    },
                    {
                        "name": "https://tools.ietf.org/html/rfc7009#section-2.2.1",
                        "refsource": "MISC",
                        "url": "https://tools.ietf.org/html/rfc7009#section-2.2.1"
                    }
                ]
            },
            "source": {
                "advisory": "GHSA-7mqr-2v3q-v2wm",
                "discovery": "UNKNOWN"
            }
        },
        "gitlab.com": {
            "advisories": [
                {
                    "affected_range": "<0.34.0",
                    "affected_versions": "All versions before 0.34.0",
                    "cvss_v2": "AV:N/AC:H/Au:N/C:P/I:P/A:N",
                    "cvss_v3": "CVSS:3.1/AV:N/AC:H/PR:N/UI:R/S:C/C:H/I:H/A:N",
                    "cwe_ids": [
                        "CWE-1035",
                        "CWE-755",
                        "CWE-937"
                    ],
                    "date": "2021-05-24",
                    "description": "In ORY Fosite (the security first OAuth2 & OpenID Connect framework for Go) before version 0.34.0, the `TokenRevocationHandler` ignores errors coming from the storage. This can lead to unexpected 200 status codes indicating successful revocation while the token is still valid. Whether an attacker can use this for her advantage depends on the ability to trigger errors in the store. This is fixed in version 0.34.0",
                    "fixed_versions": [
                        "0.34.0"
                    ],
                    "identifier": "CVE-2020-15223",
                    "identifiers": [
                        "GHSA-7mqr-2v3q-v2wm",
                        "CVE-2020-15223"
                    ],
                    "not_impacted": "All versions starting from 0.34.0",
                    "package_slug": "go/github.com/ory/fosite",
                    "pubdate": "2021-05-24",
                    "solution": "Upgrade to version 0.34.0 or above.",
                    "title": "Improper Handling of Exceptional Conditions",
                    "urls": [
                        "https://github.com/ory/fosite/security/advisories/GHSA-7mqr-2v3q-v2wm",
                        "https://nvd.nist.gov/vuln/detail/CVE-2020-15223",
                        "https://github.com/ory/fosite/commit/03dd55813f5521985f7dd64277b7ba0cf1441319",
                        "https://tools.ietf.org/html/rfc7009#section-2.2.1",
                        "https://github.com/advisories/GHSA-7mqr-2v3q-v2wm"
                    ],
                    "uuid": "059015b1-0586-44fd-9d30-a728fdd1f5fa"
                },
                {
                    "affected_range": "<0.34.0",
                    "affected_versions": "All versions before 0.34.0",
                    "cvss_v2": "AV:N/AC:H/Au:N/C:P/I:P/A:N",
                    "cvss_v3": "CVSS:3.1/AV:N/AC:H/PR:N/UI:R/S:C/C:H/I:H/A:N",
                    "cwe_ids": [
                        "CWE-1035",
                        "CWE-755",
                        "CWE-937"
                    ],
                    "date": "2021-05-24",
                    "description": "In ORY Fosite (the security first OAuth2 & OpenID Connect framework for Go) before version 0.34.0, the `TokenRevocationHandler` ignores errors coming from the storage. This can lead to unexpected 200 status codes indicating successful revocation while the token is still valid. Whether an attacker can use this for her advantage depends on the ability to trigger errors in the store. This is fixed in version 0.34.0",
                    "fixed_versions": [
                        "0.34.0"
                    ],
                    "identifier": "CVE-2020-15223",
                    "identifiers": [
                        "GHSA-7mqr-2v3q-v2wm",
                        "CVE-2020-15223"
                    ],
                    "not_impacted": "All versions starting from 0.34.0",
                    "package_slug": "go/github.com/ory/fosite/handler/oauth2",
                    "pubdate": "2021-05-24",
                    "solution": "Upgrade to version 0.34.0 or above.",
                    "title": "Improper Handling of Exceptional Conditions",
                    "urls": [
                        "https://github.com/ory/fosite/security/advisories/GHSA-7mqr-2v3q-v2wm",
                        "https://nvd.nist.gov/vuln/detail/CVE-2020-15223",
                        "https://github.com/ory/fosite/commit/03dd55813f5521985f7dd64277b7ba0cf1441319",
                        "https://github.com/advisories/GHSA-7mqr-2v3q-v2wm"
                    ],
                    "uuid": "d9f757f5-a1e5-4532-bb03-e443e9a5180d"
                }
            ]
        },
        "nvd.nist.gov": {
            "configurations": {
                "CVE_data_version": "4.0",
                "nodes": [
                    {
                        "children": [],
                        "cpe_match": [
                            {
                                "cpe23Uri": "cpe:2.3:a:ory:fosite:*:*:*:*:*:*:*:*",
                                "cpe_name": [],
                                "versionEndExcluding": "0.34.0",
                                "vulnerable": true
                            }
                        ],
                        "operator": "OR"
                    }
                ]
            },
            "cve": {
                "CVE_data_meta": {
                    "ASSIGNER": "security-advisories@github.com",
                    "ID": "CVE-2020-15223"
                },
                "data_format": "MITRE",
                "data_type": "CVE",
                "data_version": "4.0",
                "description": {
                    "description_data": [
                        {
                            "lang": "en",
                            "value": "In ORY Fosite (the security first OAuth2 & OpenID Connect framework for Go) before version 0.34.0, the `TokenRevocationHandler` ignores errors coming from the storage. This can lead to unexpected 200 status codes indicating successful revocation while the token is still valid. Whether an attacker can use this for her advantage depends on the ability to trigger errors in the store. This is fixed in version 0.34.0"
                        }
                    ]
                },
                "problemtype": {
                    "problemtype_data": [
                        {
                            "description": [
                                {
                                    "lang": "en",
                                    "value": "CWE-755"
                                }
                            ]
                        }
                    ]
                },
                "references": {
                    "reference_data": [
                        {
                            "name": "https://github.com/ory/fosite/commit/03dd55813f5521985f7dd64277b7ba0cf1441319",
                            "refsource": "MISC",
                            "tags": [
                                "Patch",
                                "Third Party Advisory"
                            ],
                            "url": "https://github.com/ory/fosite/commit/03dd55813f5521985f7dd64277b7ba0cf1441319"
                        },
                        {
                            "name": "https://tools.ietf.org/html/rfc7009#section-2.2.1",
                            "refsource": "MISC",
                            "tags": [
                                "Third Party Advisory"
                            ],
                            "url": "https://tools.ietf.org/html/rfc7009#section-2.2.1"
                        },
                        {
                            "name": "https://github.com/ory/fosite/security/advisories/GHSA-7mqr-2v3q-v2wm",
                            "refsource": "CONFIRM",
                            "tags": [
                                "Third Party Advisory"
                            ],
                            "url": "https://github.com/ory/fosite/security/advisories/GHSA-7mqr-2v3q-v2wm"
                        }
                    ]
                }
            },
            "impact": {
                "baseMetricV2": {
                    "acInsufInfo": false,
                    "cvssV2": {
                        "accessComplexity": "HIGH",
                        "accessVector": "NETWORK",
                        "authentication": "NONE",
                        "availabilityImpact": "NONE",
                        "baseScore": 4.0,
                        "confidentialityImpact": "PARTIAL",
                        "integrityImpact": "PARTIAL",
                        "vectorString": "AV:N/AC:H/Au:N/C:P/I:P/A:N",
                        "version": "2.0"
                    },
                    "exploitabilityScore": 4.9,
                    "impactScore": 4.9,
                    "obtainAllPrivilege": false,
                    "obtainOtherPrivilege": false,
                    "obtainUserPrivilege": false,
                    "severity": "MEDIUM",
                    "userInteractionRequired": true
                },
                "baseMetricV3": {
                    "cvssV3": {
                        "attackComplexity": "HIGH",
                        "attackVector": "NETWORK",
                        "availabilityImpact": "NONE",
                        "baseScore": 8.0,
                        "baseSeverity": "HIGH",
                        "confidentialityImpact": "HIGH",
                        "integrityImpact": "HIGH",
                        "privilegesRequired": "NONE",
                        "scope": "CHANGED",
                        "userInteraction": "REQUIRED",
                        "vectorString": "CVSS:3.1/AV:N/AC:H/PR:N/UI:R/S:C/C:H/I:H/A:N",
                        "version": "3.1"
                    },
                    "exploitabilityScore": 1.6,
                    "impactScore": 5.8
                }
            },
            "lastModifiedDate": "2020-10-05T18:22Z",
            "publishedDate": "2020-09-24T17:15Z"
        }
    }
}